### PR TITLE
Dasharo OVMF: ESP Partition Scanning in look for grubx64.efi or shimx64.efi

### PR DESCRIPTION
**Pull request by maheshtammisetti**
Written a function according to the mentioned issue enhancement for the ESP Partition Scanning.

### CODE REVIEW

**Comment by miczyg1** [comment]
These changes were already merged here https://github.com/Dasharo/edk2/pull/44/

Please update your local tree @maheshtammisetti

**Comment by maheshtammisetti** [comment]
Yes, I see that the issue is with the local tree. @miczyg1 My changes in BmBoot.c is visible?

**Comment by miczyg1** [comment]
Yup, visible

**Comment by macpijan** [comment]
@maheshtammisetti You can see what is visible for everyone by checking the MR first https://github.com/Dasharo/edk2/pull/60/files

**Comment by maheshtammisetti** [comment]
`/home/coreboot/coreboot/OvmfPkg/OvmfPkgX64.dsc(588): error F001: Pcd (gDasharoSystemFeaturesTokenSpaceGuid.PcdDasharoEnterprise) defined in DSC is not declared in DEC files referenced in INF files in FDF. Arch: ['X64']`  - @miczyg1 error still persists even after the local tree update, perhaps PCD isn't really declared in the DEC?

**Comment by miczyg1** [comment]
> `/home/coreboot/coreboot/OvmfPkg/OvmfPkgX64.dsc(588): error F001: Pcd (gDasharoSystemFeaturesTokenSpaceGuid.PcdDasharoEnterprise) defined in DSC is not declared in DEC files referenced in INF files in FDF. Arch: ['X64']` - @miczyg1 error still persists even after the local tree update, perhaps PCD isn't really declared in the DEC?

https://github.com/Dasharo/DasharoModulePkg/blob/main/DasharoModulePkg.dec#L50

**Comment by miczyg1** [comment]
@maheshtammisetti GH says there is still a conflict in OvmfPkg/OvmfPkgX64.dsc, so you definitely still have problems with local tree

**Comment by maheshtammisetti** [comment]
Let me look into it

File: MdeModulePkg/Library/UefiBootManagerLib/BmBoot.c [hunk 1]
```diff
@@ -2639,6 +2639,171 @@ BmEnumerateBootOptions (
   return BootOptions;
 }
 
+/* EFI System partition Scanning code Extension */
+
+EFI_BOOT_MANAGER_LOAD_OPTION * BmEspScanning (
+
+UINTN							*BootOptionCount,
+				
+OUT   EFI_BOOT_MANAGER_LOAD_OPTION			*BootOption,
```
Comment (miczyg1) [inline MdeModulePkg/Library/UefiBootManagerLib/BmBoot.c:2648]: Please have a look at the codestyle in this file.  Your added code does not comply with EDK2 code style at all. Please reformat it @maheshtammisetti

Comment (maheshtammisetti) [reply MdeModulePkg/Library/UefiBootManagerLib/BmBoot.c:2648]: Yes, I was aware after I have seen it in the GH, and I am working on it right now.

Comment (maheshtammisetti) [reply MdeModulePkg/Library/UefiBootManagerLib/BmBoot.c:2648]: @miczyg1 `/home/coreboot/coreboot/DasharoModulePkg/DasharoBootPolicies/DasharoBootPolicies.inf(-1): error 000E: File/directory not found in workspace
	/home/coreboot/coreboot/IntelSiliconPkg/IntelSiliconPkg.dec`, I have updated the local repository from the main for the DasharoModulePkg, but this is a bit confusing.

Comment (miczyg1) [reply MdeModulePkg/Library/UefiBootManagerLib/BmBoot.c:2648]: I have just built OVMF from dasharo branch using instructions from https://docs.dasharo.com/variants/qemu_q35/building-manual/

Please ensure you are following the same process

Comment (maheshtammisetti) [reply MdeModulePkg/Library/UefiBootManagerLib/BmBoot.c:2648]: Yes sure, I even looked at the dasharo branch and will be doing the same now

**Comment by miczyg1** [comment]
Implemented in https://github.com/Dasharo/edk2/pull/70

**Pull request closed without merging**